tiatoolbox/dataloader/wsireader.py

```diff
@@ -110,8 +244,26 @@ def slide_thumbnail(self):
         Returns:
             ndarray : image array
 
+        Examples:
+            >>> from tiatoolbox.dataloader import wsireader
+            >>> wsi_obj = wsireader.OpenSlideWSIReader(input_dir="./",
+            ...     file_name="CMU-1.ndpi")
+            >>> slide_thumbnail = wsi_obj.slide_thumbnail()
+
         """
-        raise NotImplementedError
+        level, post_read_scale = self.optimal_relative_level_scale(scale, units)
+        level_dimensions = self.slide_info.level_dimensions[level]
+        thumb = self.read_region(0, 0, *level_dimensions)
+
+        if scale != 1.0:
+            new_size = np.round(np.array(level_dimensions) * post_read_scale)
+            new_size = tuple(new_size.astype(int))
+            thumb = cv2.resize(thumb, new_size, interpolation=cv2.INTER_AREA)
+            cv2.imwrite("/home/john/Downloads/thumbnail.png", thumb)
```

Comment:
I think this will generate an error on some machines. This should be a path specified by user.

Reply:
Oops, looks like a line for debugging snuck in there!